zulipterminal/model.py

```diff
@@ -853,6 +827,11 @@ def _update_topic_index(self, stream_id: int, topic_name: str) -> None:
         Update topic order in index based on incoming message.
         Helper method called by _handle_message_event
         """
+        # Fetch topics before updating the index if they have not been fetched
+        # yet via the topic toggle.
+        if not self.index['topics'][stream_id]:
+            self.get_topics_in_stream([stream_id])
```

Comment:
This is not specific to your change here, but reading this code reminds me that `get_` is unclear just from the name - this is actually an API call, whereas `get_` can express any case where you're gathering data through an object, from a resource or otherwise (particularly if you come from languages where "getters" and "setters" are standard). In this context I might expect any external access of the topic list for a stream to be hidden behind a `get_*` (or no prefix), with any potential API call (`fetch_*`) generally happening behind the scenes.

To clarify, this is really another issue, since we do this in a few cases.

Reply:
Agreed. It would make it easier to depict what the method actually does.